plugin/trino-ldap-group-provider/src/test/java/io/trino/plugin/ldapgroup/TestLdapGroupProviderIntegration.java

```diff
@@ -0,0 +1,314 @@
+                "ldap.user-base-dn", "ou=users,dc=trino,dc=testldap,dc=com",
+                "ldap.user-search-filter", "(cn={0})");
+
+        DisposableSubContext groupsOU = openLdapServer.createOrganization("groups");
+        DisposableSubContext externalGroupsOU = openLdapServer.createOrganization("external", groupsOU);
+        DisposableSubContext usersOU = openLdapServer.createOrganization("users");
+
+        DisposableSubContext johnb = openLdapServer.createUser(usersOU, "johnb", ImmutableMap.of(
+                "employeeNumber", "1000",
+                "mail", "johnb@company.com"));
+        DisposableSubContext alicea = openLdapServer.createUser(usersOU, "alicea", ImmutableMap.of(
+                "employeeNumber", "1001",
+                "mail", "alicea@trino.io",
+                "givenname", "Alice"));
+        DisposableSubContext bobq = openLdapServer.createUser(usersOU, "bobq", ImmutableMap.of(
+                "employeeNumber", "1002"));
+        openLdapServer.createUser(usersOU, "carlp", ImmutableMap.of(
+                "employeeNumber", "1003"));
+
+        clients = openLdapServer.createGroup(externalGroupsOU, "clients");
+        openLdapServer.addUserToGroup(johnb, clients);
+        openLdapServer.addUserToGroup(alicea, clients);
+
+        developers = openLdapServer.createGroup(groupsOU, "developers");
+        openLdapServer.addUserToGroup(alicea, developers);
+
+        qa = openLdapServer.createGroup(groupsOU, "qa");
+        openLdapServer.addUserToGroup(alicea, qa);
+        openLdapServer.addUserToGroup(bobq, qa);
+    }
+
+    @AfterAll
+    public void close()
+            throws Exception
+    {
+        closer.close();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideUsersAndExpectedGroups")
+    public void testGetGroups(ConfigBuilder configBuilder, String userName, Set<String> expectedGroups)
+    {
+        Map<String, String> config = configBuilder.apply(ImmutableMap.<String, String>builder().putAll(baseConfig)).buildOrThrow();
+        GroupProvider groupsProvider = factory.create(config);
+
+        Set<String> groups = groupsProvider.getGroups(userName);
+
+        assertEquals(expectedGroups, groups);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideForTestGetGroupsAndUserAttributes")
+    public void testGetGroupsWithUserAttributes(
+            ConfigBuilder configBuilder,
+            String userName,
+            String searchAttribute,
+            Set<String> expectedGroups)
+    {
+        Map<String, String> config = configBuilder.apply(
+                ImmutableMap.<String, String>builder()
+                    .putAll(baseConfig)
+                    .put("ldap.user-search-attributes", searchAttribute))
+                .buildOrThrow();
+        GroupProvider groupsProvider = factory.create(config);
+
+        Set<String> groups = groupsProvider.getGroups(userName);
+
+        assertEquals(expectedGroups, groups);
+    }
+
+    @ParameterizedTest
```

Comment:
Can we avoid using `ParametermizedTest`, it makes harder to debug the tests.

Reply:
The integrations tests already use `ParametermizedTest` to get a cross-product of configurations to tests (e.g., using `ldap.user-member-of-attribute` or `ldap.group-base-dn`) - I don't see any better way of doing this